crypto-app/src/pages/Coins/Coins.js

```diff
@@ -0,0 +1,131 @@
+  getCoins = async () => {
+    try {
+      this.setState({
+        isLoading: true
+      })
+      const { data } = await axios(
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+      );
+      this.setState({
+        coins: data,
+        isLoading: false,
+      });
+    } catch(error) {
+      this.setState({
+        isLoading: false,
+        hasError: true,
+        errMess: "There was a problem getting coin list!"
+      })
+    }
+  }
+  getMoreCoins = async () => {
+    try {
+      this.setState({
+        page: this.state.page + 1,
+        isLoading: true,
+      });
+      const  { data }  = await axios(
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=${this.state.page}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+      );
+      const addCoins = [...this.state.coins, ...data]
+      setTimeout(() => {
+        this.setState({
+          coins: addCoins,
+          isLoading: false
+        })
+      }, 500)
+    }catch(error){
+      this.setState({isLoading: false, errMess:"Could not load more coins!"})
+    }
+  }
+  componentDidMount() {
+      this.getCoins()
+  }
+  render(){
+    return (
+      <>
+        <TableContainer>
+          <InfiniteScroll
+            dataLength={this.state.coins.length}
+            next={this.getMoreCoins}
+            hasMore={true}
+            loader={(this.state.isLoading && <h4>Loading...</h4>) || <h4>{this.state.errMess}</h4>}
+          >
+            <Table>
+              <thead>
+                <TableRow>
+                  <TableHeader>#</TableHeader>
+                  <TableHeader>Name</TableHeader>
+                  <TableHeader>Price</TableHeader>
+                  <TableHeader>1h%</TableHeader>
+                  <TableHeader>24h%</TableHeader>
+                  <TableHeader>7d%</TableHeader>
+                  <TableHeader>24hr Volume/Market Cap</TableHeader>
+                  <TableHeader>Circulating/Total Supply</TableHeader>
+                  <TableHeader>Last 7d</TableHeader>
+                </TableRow>
+              </thead>
+              <tbody>
+                {(this.state.coins && this.state.coins.map((coin) => (
+                  <CoinInstance
+                    key={coin.id}
+                    name={coin.name}
+                    price={
+                      coin.current_price < 1e3
+                        ? coin.current_price.toString()
+                        : coin.current_price.toLocaleString()
+                    }
+                    image={coin.image}
+                    symbol={coin.symbol.toUpperCase()}
+                    oneHour={
+                      coin.price_change_percentage_1h_in_currency === null
+                        ? null
+                        : coin.price_change_percentage_1h_in_currency.toString()
+                    }
```

Comment:
`oneHour={coin.price_change_percentage_1h_in_currency?.toString()}`

this might help